### ISSUE TITLE
Doctrine\Common\Annotations\AnnotationException [Semantical Error] The annotation "@Ray\Di\Di\Inject" does not exist, or could not be auto-loaded

### ISSUE THREAD

**Issue by vanoostrum** [ray-di/Ray.Di#164]
I'm getting this error after upgrading ray/di from 2.5.3 to 2.6.1. I'm not sure whether the error lies in ray/di or in doctrine, but since it happened after upgrading ray/di I'm posting the issue here. I assume there were no breaking changes from 2.5 to 2.6, am I correct? If there are breaking changes then I should probably file an issue for lorenzo/piping-bag. (https://github.com/lorenzo/piping-bag)

Stacktrace (CakePHP application with lorenzo/piping-bag which uses ray/di):

```
Doctrine\Common\Annotations\AnnotationException [Semantical Error] The annotation "@Ray\Di\Di\Inject" in method App\Controller\AppController::setSession() does not exist, or could not be auto-loaded.
	#0 vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php(734): Doctrine\Common\Annotations\AnnotationException->semanticalError(The annotation "@Ray\Di\Di\Inject" in method App\Controller\AppController::setSession() does not exist, or could not be auto-loaded.)
	#1 vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php(663): Doctrine\Common\Annotations\DocParser->Annotation()
	#2 vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/DocParser.php(354): Doctrine\Common\Annotations\DocParser->Annotations()
	#3 vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationReader.php(284): Doctrine\Common\Annotations\DocParser->parse(/**
 * @Inject
 * @param Session $session
 */, method App\Controller\AppController::setSession())
	#4 vendor/doctrine/annotations/lib/Doctrine/Common/Annotations/AnnotationReader.php(292): Doctrine\Common\Annotations\AnnotationReader->getMethodAnnotations(ReflectionMethod)
	#5 vendor/ray/di/src/AnnotatedClassMethods.php(58): Doctrine\Common\Annotations\AnnotationReader->getMethodAnnotation(ReflectionMethod, Ray\Di\Di\InjectInterface)
	#6 vendor/ray/di/src/AnnotatedClass.php(46): Ray\Di\AnnotatedClassMethods->getSetterMethod(ReflectionMethod)
	#7 vendor/ray/di/src/DependencyFactory.php(25): Ray\Di\AnnotatedClass->getNewInstance(ReflectionClass)
	#8 vendor/ray/di/src/Untarget.php(36): Ray\Di\DependencyFactory->newAnnotatedDependency(ReflectionClass)
	#9 vendor/ray/di/src/Bind.php(68): Ray\Di\Untarget->__invoke(Ray\Di\Container, Ray\Di\Bind)
	#10 vendor/ray/di/src/Injector.php(83): Ray\Di\Bind->__destruct()
	#11 vendor/ray/di/src/Injector.php(71): Ray\Di\Injector->bind(App\Controller\PagesController)
	#12 vendor/lorenzo/piping-bag/src/Di/PipingBag.php(83): Ray\Di\Injector->getInstance(App\Controller\PagesController, )
	#13 vendor/lorenzo/piping-bag/src/Routing/Filter/ControllerFactoryFilter.php(63): PipingBag\Di\PipingBag->get(App\Controller\PagesController)
	#14 vendor/cakephp/cakephp/src/Routing/Filter/ControllerFactoryFilter.php(49): PipingBag\Routing\Filter\ControllerFactoryFilter->_getController(Cake\Network\Request, Cake\Network\Response)
	#15 vendor/cakephp/cakephp/src/Routing/DispatcherFilter.php(144): Cake\Routing\Filter\ControllerFactoryFilter->beforeDispatch(Cake\Event\Event)
	#16 vendor/cakephp/cakephp/src/Event/EventManager.php(426): Cake\Routing\DispatcherFilter->handle(Cake\Event\Event, Cake\Network\Request, Cake\Network\Response)
	#17 vendor/cakephp/cakephp/src/Event/EventManager.php(391): Cake\Event\EventManager->_callListener(array, Cake\Event\Event)
	#18 vendor/cakephp/cakephp/src/Event/EventDispatcherTrait.php(78): Cake\Event\EventManager->dispatch(Cake\Event\Event)
	#19 vendor/cakephp/cakephp/src/Http/ActionDispatcher.php(80): Cake\Http\ActionDispatcher->dispatchEvent(Dispatcher.beforeDispatch, array)
	#20 vendor/cakephp/cakephp/src/Http/BaseApplication.php(83): Cake\Http\ActionDispatcher->dispatch(Cake\Network\Request, Cake\Network\Response)
	#21 vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Http\BaseApplication->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response, Cake\Http\Runner)
	#22 vendor/cakephp/cakephp/src/Routing/Middleware/RoutingMiddleware.php(62): Cake\Http\Runner->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response)
	#23 vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Routing\Middleware\RoutingMiddleware->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response, Cake\Http\Runner)
	#24 vendor/cakephp/cakephp/src/Routing/Middleware/AssetMiddleware.php(93): Cake\Http\Runner->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response)
	#25 vendor/cakephp/cakephp/src/Http/Runner.php(65): Cake\Routing\Middleware\AssetMiddleware->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response, Cake\Http\Runner)
	#26 vendor/xel-webservices/xel-cake-commons/src/Cake/Middleware/AppErrorHandlerMiddleware.php(66): Cake\Http\Runner->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response)
	#27 vendor/cakephp/cakephp/src/Http/Runner.php(65): Xel\Cake\Middleware\AppErrorHandlerMiddleware->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response, Cake\Http\Runner)
	#28 vendor/cakephp/cakephp/src/Http/Runner.php(51): Cake\Http\Runner->__invoke(Zend\Diactoros\ServerRequest, Zend\Diactoros\Response)
	#29 vendor/cakephp/cakephp/src/Http/Server.php(90): Cake\Http\Runner->run(Cake\Http\MiddlewareQueue, Zend\Diactoros\ServerRequest, Zend\Diactoros\Response)
	#30 webroot/index.php(37): Cake\Http\Server->run()
```

This is the code: 

```
namespace App\Controller;

use Ray\Di\Di\Inject;

class AppController extends XelAppController {

    /** @var  Session */
    protected $session;

    /**
     * @Inject
     * @param Session $session
     */
    public function setSession(Session $session) {
        $this->session = $session;
    }

    //...
}
```

**Comment by koriym** [comment]
```
Doctrine\Common\Annotations\AnnotationException [Semantical Error] The annotation "@Ray\Di\Di\Inject" ... does not exist, or could not be auto-loaded.
```

This error is just as stated, `@Inject` annotation is not loadable. (No autoloader is registered for `Inject.php` annotation class)

**Comment by koriym** [comment]
Try

```php
use Doctrine\Common\Annotations\AnnotationRegistry;
$loader = require dirname(dirname(__DIR__)) . '/vendor/autoload.php'; // composer autoload.php
AnnotationRegistry::registerLoader([$loader, 'loadClass']);
```
or

```php
use Doctrine\Common\Annotations\AnnotationRegistry;
AnnotationRegistry::registerLoader('class_exists');
```

in somewhere your bootstrap script.

**Comment by koriym** [comment]
I found the actual cause. I try to find the way to avoid BC break.

**Comment by koriym** [comment]
Released as 2.6.2
https://github.com/ray-di/Ray.Di/releases/tag/2.6.2

**Comment by vanoostrum** [comment]
Yes it's working with the new version, thanks for the quick response!
